src/main/java/com/scottlogic/sorts/AuthorPostSorter.java

```diff
@@ -0,0 +1,32 @@
+package com.scottlogic.sorts;
+
+import com.scottlogic.NullPostChecker;
+import com.scottlogic.UserPost;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Comparator.*;
+
+public class AuthorPostSorter implements PostSorter {
+
+
+    @Override
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder orderIn) {
+
+        List<UserPost> listToSort = new ArrayList<>();
+        if (inputList == null) {
+            return listToSort;
+        }
+        listToSort = NullPostChecker.nullPostCheck(inputList);
+
+        switch (orderIn) {
+
+            case ASC -> Collections.sort(listToSort,comparing(UserPost::getAuthor, nullsLast(naturalOrder())));
+
+            case DESC -> Collections.sort(listToSort, Collections.reverseOrder(comparing(UserPost::getAuthor, nullsFirst(naturalOrder()))));
```

Comment:
Nice bit of tidy-up here. We'll get it even tidier when we move on to looking at streams!